ZFSin/zfs/lib/libzfs/libzfs_sendrecv.c

```diff
@@ -522,7 +503,11 @@ cksummer(void *arg)
 	umem_cache_destroy(ddt.ddecache);
 	free(ddt.dedup_hash_array);
 	free(buf);
-#ifndef WIN32
+#ifdef _WIN32
+	// We don't want to wrap fclose() so this is a special
+	// place we manually close the handle from socketpair().
+	CloseHandle(ofp);
+#else
 	(void) fclose(ofp);
```

Comment:
And the matching special case for `SOCKET` close. We could call `close()` / `wosix_close()` here (I did make that special code to detect sockets, why did I do that then not use it). But it seems a bit unclear to mix `fclose()` and `close()`.